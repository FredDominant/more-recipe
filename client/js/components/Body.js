import React from 'react';
import ReactDOM from 'react-dom';
import { Switch, Route } from 'react-router-dom';

import { Home } from './Home';
import { Profile } from './Profile';
import { Recipe } from './Recipe'
import { UserHome } from './UserHome';

export class Body extends React.Component {
  render() {
    return (
			<Switch>
				<Route path='/' exact component={Home} />
				<Route path='/home' exact component={UserHome} />
				<Route path='/recipe' component={Recipe} />
				<Route path='/profile' component={Profile} />
			</Switch>
    );
  }
}
