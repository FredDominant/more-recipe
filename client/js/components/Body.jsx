import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from './Home';
import Profile from './Profile';
import Recipe from './Recipe';
import UserHome from './UserHome';

const Body = () => (
  <Switch>
    <Route path="/" exact component={Home} />
    <Route path="/home" exact component={UserHome} />
    <Route path="/profile" component={Profile} />
    <Route path="/recipe/:recipeId" component={Recipe} />
    <Route path="/user/:userId" component={Profile} />
    <Route component={Recipe} />
  </Switch>
);
export default Body;
