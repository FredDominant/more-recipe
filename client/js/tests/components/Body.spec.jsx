import React from 'react';
import { shallow } from 'enzyme';

import Body from '../../components/Body';
/**
 * @description test for Body component
 *
 * @returns {null} null
 */
function setup() {
  const wrapper = shallow(<Body />);
  return {
    wrapper
  };
}

describe('Test for Body component', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
