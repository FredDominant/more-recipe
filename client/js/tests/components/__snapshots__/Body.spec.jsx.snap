// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Body component should render correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/add-recipe"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/profile"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/favourites"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/user/recipes"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/recipe/:recipeId"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/search"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/recipe/edit/:recipeId"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/user/password-reset/:token"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
